### ISSUE TITLE
TG2-VALIDATION_GEOGRAPHY_STANDARD

### ISSUE THREAD

**Issue by chicoreus** [tdwg/bdq#139]
| TestField | Value |
| ----- | ----- |
| **GUID** | 9d6f53c0-775b-4579-b7a4-5e5f093aa512 |
| **Label** | VALIDATION_GEOGRAPHY_STANDARD |
| **Description** | Can the individual values of the terms dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality be unambiguously resolved from bdq:sourceAuthority? |
| **TestType** | Validation |
| **Darwin Core Class** | Location |
| **Information Elements ActedUpon** | dwc:continent |
|  | dwc:country |
|  | dwc:countryCode |
|  | dwc:stateProvince |
|  | dwc:county |
|  | dwc:municipality |
| **Information Elements Consulted** |  |
| **Expected Response** | EXTERNAL_PREREQUISITES_NOT_MET if the bdq:sourceAuthority is not available; INTERNAL_PREREQUISITES_NOT_MET if all of the terms dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality are EMPTY; COMPLIANT if the individual values of dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality can be unambiguously resolved from the bdq:sourceAuthority; otherwise NOT_COMPLIANT |
| **Data Quality Dimension** | Conformance |
| **Term-Actions** | GEOGRAPHY_STANDARD |
| **Parameter(s)** | bdq:sourceAuthority |
| **Source Authority** | bdq:sourceAuthority default = "The Getty Thesaurus of Geographic Names (TGN)" [https://www.getty.edu/research/tools/vocabularies/tgn/index.html] |
| **Specification Last Updated** | 2023-09-22 |
| **Example** | dwc:continent="Oceania", dwc:country="Australia", dwc:stateProvince="Virginia" |
| **Source** | VertNet, Kurator |
| **References** | <ul><li>Chapman, AD and Wieczorek, JR (2020). Georeferencing Best Practices. Copenhagen: GBIF Secretariat (https://doi.org/10.15468/doc-gg7h-s853)</li> <li>VertNet (2019). DwCVocabs (https://github.com/VertNet/DwCVocabs/blob/master/vocabs/Geography.csv)</li> <li>Getty Reseach Institute (2017). Getty Thesaurus of Geographic Names Online (https://www.getty.edu/research/tools/vocabularies/tgn/index.html)</li></ul>  |
| **Example Implementations (Mechanisms)** | Kurator |
| **Link to Specification Source Code** | https://github.com/VertNet/toolkit, https://github.com/kurator-org/kurator-validation/blob/master/packages/kurator_dwca/workflows/dwca_geography_assessor.yaml |
| **Notes** | This test only tests if the values of the Information Elements values can be found in the Source Authority. Test #95 tests the combination of these values. Only the administrative terms are considered.  Terms for waterBody and islands are not included in this test. |

**Comment by chicoreus** [comment]
The discussion in #118 suggests the need for this validation. Added here for discussion.

**Comment by ArthurChapman** [comment]
@chicoreus - does this mean that all higher geography terms need to be unambiguously resolvable.  What about if you only have down to country and nothing lower?

**Comment by ArthurChapman** [comment]
@chicoreus question has not been answered.  Is it COMPLIANT if some can be unambiguously resolved - or does all have to be unambiguously resolvable

**Comment by tucotuco** [comment]
I feel strongly that this test must be that the entire higher geography combination is unambiguous, as it is stated. It is not really functional to do a piece-wise validation of the geographic terms and posit compliance. If you want to go the route of checking to, for example, country, I believe this test would have to change to add another parameter to say how many levels down you want to be checked. Even so, the test would not check continent, then country, then stateProvince independently - it is only the combination that makes sense to be compliant.

**Comment by Tasilee** [comment]
@tucotuco - Are you are saying that the test would return NOT_COMPLIANT if the not EMPTY terms among dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county and dwc:municipality resulted in ambiguity? If so, it raises an interesting point that the ambiguity needs to be reported?

**Comment by tucotuco** [comment]
The test isn't about ambiguity, it is about being non-standard against a given authority. This is a NOTSTANDARD test, not a INCONSISTENT test. In NOTSTANDARD tests we don't say why it is not standard. Indeed, we don't even make that the job of consistency tests (e.g., https://github.com/tdwg/bdq/issues/67) except to test for something very specific (e.g., https://github.com/tdwg/bdq/issues/62).

**Comment by Tasilee** [comment]
@tucotuco ok. tare

**Comment by ArthurChapman** [comment]
I just noticed that the wording of this test is virtually identical to #95 - they appear duplicates!

#139
COMPLIANT if the combination of dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality can be unambiguously resolved from the bdq:sourceAuthority; 

#95 
COMPLIANT if the combination of values of administrative geographic terms (dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality) can be unambiguously resolved by the bdq:sourceAuthority;

**Comment by Tasilee** [comment]
I was going to use "administrative geographic terms" in the Description of one of these tests and decided to just be explicit, as we have opted for elsewhere.

**Comment by ArthurChapman** [comment]
But are the tests the same, and can one be deleted?

**Comment by Tasilee** [comment]
I think you are right. They are close enough.

**Comment by ArthurChapman** [comment]
@tucotuco Can you check those two tests and see if there is any reason we have two tests (#139 and #95)

**Comment by Tasilee** [comment]
#139: COMPLIANT if the combination of [values] of dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality can be unambiguously resolved from the bdq:sourceAuthority;

#95: COMPLIANT if the combination of values of dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality can be unambiguously resolved by the bdq:sourceAuthority

So....there are equal in intent. I suggest we CLOSE #95 and I'll edit this one to add "values of". 

Votes please?

**Comment by tucotuco** [comment]
Sorry folks. These two tests are not supposed to be the same. This one is not about ambiguity, while #95 is. The geography can be not standard and unambiguous.

I could see having only one of these two tests, but I think #95 is actually the useful one in that case.

**Comment by Tasilee** [comment]
Thanks @tucotuco. Your call. I guess the questions are then

- Are the potential combinations of STANDARD/NOT_STANDARD | AMBIGUOUS/UNAMBIGUOUS informative?
- If they are, then the Expected Responses need to more clearly differentiate the tests.
- If not (which you simply by suggesting that #139 may be dropped), fine.

**Comment by chicoreus** [comment]
The examples seem informative.  #95 dwc:stateProvince="WA"   Is standard, but ambiguous, other terms would be needed to disambiguate which WA.   #139  dwc:continent="Oceania", dwc:country="Australia", dwc:stateProvince="Virginia" is not standard, there is no Virginia within Oceania:Australia, no such higher geography exists, while at least two WA higher geographies exist.

The whiteboard diagram from Ganesville on #95 seems to suggest that #95 was being thought of as encompasing both of these at that time.

**Comment by tucotuco** [comment]
Yes, I am having a hard time deciding if #139 is necessary. #95 is the
useful one in my opinion.

**Comment by ArthurChapman** [comment]
Is there a way we can reword one of them to cover both cases.  It is obvious that over time the two tests that were distinct at one time have converged over time.  As @chicoreus suggests, the examples are helpful.

**Comment by tucotuco** [comment]
I think they are very different and an attempt should not be made to merge them. The fact that the Expected Responses look the same is an accident. I think the correct Expected Response for this step should be something more akin to:

EXTERNAL_PREREQUISITES_NOT_MET if the bdq:sourceAuthority is not available; INTERNAL_PREREQUISITES_NOT_MET if all of the terms dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality are EMPTY; COMPLIANT if each of the NOT_EMPTY values of dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality is a STANDARD value for that term and is consistent with all of the other values in the listed fields according to the bdq:sourceAuthority; otherwise NOT_COMPLIANT

There are problems inherent in what standard means here that make me think this test is not as useful as #95.

I think I am convincing myself to drop this one.

**Comment by Tasilee** [comment]
We need to make a decision on #139 and #95. Hold till the next Zoom or 'bite (bight? :) the bullet'?

**Comment by tucotuco** [comment]
I remain fine with omitting #139.

**Comment by Tasilee** [comment]
The thumbs up suggests that #139 does not add significantly to #95 but for the deletion of a test, I'd value @chicoreus's 'vote'.

**Comment by ArthurChapman** [comment]
Because of the confusion between this test and #95, I suggest rewording as

INTERNAL_PREREQUISITES_NOT_MET if all of the terms dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality are EMPTY; COMPLIANT if the values of dwc:continent, dwc:country, dwc:countryCode, dwc:stateProvince, dwc:county, dwc:municipality can be individually resolved from the bdq:sourceAuthority; otherwise NOT_COMPLIANT

**Comment by ArthurChapman** [comment]
There has been discussion between Issues #95 and #139 - the wording converged over time such as the two tests appeared to be testing for the same thing.  Discussion on ZOOM has resulted in separating the two.  

#139 is testing individual terms for validity at that level - it looks at only one level in the hierarchy at a time and checks the validity of what is there at the level.  Hence the suggested change in wording above.
#95 is testing for inconsistencies between levels in the hierarchy - for example Western Australia (WA) as a State, and USA as a country - i.e. one is wrong and thus ambiguous.

**Comment by Tasilee** [comment]
The zoom discussion with @ArthurChapman, @tucotuco and @chicoreus today concluded that tests https://github.com/tdwg/bdq/issues/95, https://github.com/tdwg/bdq/issues/139 and https://github.com/tdwg/bdq/issues/118 were going to be very difficult to implement properly given the lack of a consistent geographic terms hierarchy by comparison with the taxonomic terms. Note the issues arising from the table above for example. We will therefore remove these tests from CORE.

In their place, we will

1. Add a test for dwc:stateProvice found to complement https://github.com/tdwg/bdq/issues/21 (which we will rename)
2. Add a test for dwc:country dwc:stateProvince combo exist at least once in the bdq:sourceAuthority (country-state/province consistent)
3. Add a test for dwc:country dwc:stateProvince combo exists exactly once in the bdq:sourceAuthority ((country-state/province unambiguous)

**Comment by ArthurChapman** [comment]
Splitting Information Elements into "Information Elements ActedUpon" and "Information Elements Consulted".

Changed "Field" to "TestField", "Output Type" to "TestType", deleted "Warning Type" and added TestField "Specification Last Updated"

**Comment by Tasilee** [comment]
Specifications updated to align with the current template
